web/src/features/time/HistoricalTimeHeader.tsx

```diff
@@ -11,61 +11,39 @@ import { RouteParameters } from 'types';
 import { useNavigateWithParameters } from 'utils/helpers';
 import { endDatetimeAtom, isHourlyAtom, startDatetimeAtom } from 'utils/state/atoms';
 
-function setToMidnightUTC(date: Date): Date {
-  const newDate = new Date(date);
-  newDate.setUTCHours(0, 0, 0, 0);
-  return newDate;
-}
-function formatDateToISO(date: Date): string {
-  return date.toISOString().slice(0, -5) + 'Z';
-}
+const TWENTY_FOUR_HOURS = 24 * 60 * 60 * 1000;
 
 export default function HistoricalTimeHeader() {
   const { i18n } = useTranslation();
   const startDatetime = useAtomValue(startDatetimeAtom);
   const endDatetime = useAtomValue(endDatetimeAtom);
+  const isHourly = useAtomValue(isHourlyAtom);
   const { isLoading } = useGetState();
   const { urlDatetime } = useParams<RouteParameters>();
   const navigate = useNavigateWithParameters();
-  const isHourly = useAtomValue(isHourlyAtom);
-  function handleLeftClick() {
-    if (!urlDatetime && !endDatetime) {
-      return;
-    }
 
-    const targetDate = urlDatetime ? new Date(urlDatetime) : endDatetime;
-    if (!targetDate) {
+  const handleTimeNavigation = (direction: 'left' | 'right') => {
```

Comment:
Subtracting/adding `TWENTY_FOUR_HOURS` is a lot easier to understand than previously. But I think I preferred the separate click handlers rather than passing in the direction (since we still need to create [distinct click handlers](https://github.com/electricitymaps/electricitymaps-contrib/pull/7397/commits/3fc3de50ee9d226127e1b7d9d26fae0e3bcb816e#diff-14a5cbe59e006d396425a12e6f93bc00e744e313563bb64bb23aecfb00165ab5R64) anyways). We could pull out the shared logic into functions. What do you think about something like this (I haven't actually tested this, so not sure if it's a direct translation, but something similar):
```javascript
const clampedDatetime = (date: Date): string => {
  // if date is greater than twentyFourHoursAgo, then go to latest
  if (date.getTime() >= Date.now() - TWENTY_FOUR_HOURS) {
    return '';
  }
  return date.toISOString();
};

const navigateDateOffset = (offset: number) => {
  const date = new Date(currentDate.getTime() + offset);
  navigate({ datetime: clampedDatetime(date) });
}

const handleLeftNavigation = () => navigateDateOffset(-TWENTY_FOUR_HOURS)

const handleRightNavigation = () => {
  if (!endDatetime || !urlDatetime) {
    return;
  }
  navigateDateOffset(TWENTY_FOUR_HOURS)
}
```

Reply:
Also, are we still planning on anchoring midnight to midnight?

Reply:
I think for this v1 it's better to not anchor to midnight. From a users perspective this makes the most sense without limiting the current day to the elapsed hours.

Reply:
Good point on the separate click handlers, I've pulled them apart to be completely isolated as we'll probably adjust their functionality to be even more unique in the future.